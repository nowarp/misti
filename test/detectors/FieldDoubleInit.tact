contract Test1 {
    a1: Int = 0;
    a2: Int = 0;
    init(x: Int) {
        self.a1 = x; // Should be highlighted
        // @misti:suppress FieldDoubleInit
        self.a2 = x; // OK: Suppressed
    }
}

contract Test2 {
    a: Int;
    b: Int = 42; // No false positive
    init(x: Int) {
        self.a = x; // No false positive
    }
    get fun getB(): Int { return self.b; }
}
