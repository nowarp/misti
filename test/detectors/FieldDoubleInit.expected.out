[MEDIUM] FieldDoubleInit: Field a1 is initialized twice
test/detectors/FieldDoubleInit.tact:5:9:
  4 |     init(x: Int) {
> 5 |         self.a1 = x; // Should be highlighted
              ^
  6 |         // @misti:suppress FieldDoubleInit
Help: Consider initializing the field only in its declaration or in the `init` function
See: https://nowarp.io/tools/misti/docs/detectors/FieldDoubleInit